### ISSUE TITLE
ctypes.util.find_library needlessly call crle on Solaris

### ISSUE THREAD

**Issue by dad3cdc4-bbab-4939-8082-aaf913d8272e** [python/cpython#67476]
BPO | [23287](https://bugs.python.org/issue23287)
--- | :---
Nosy | @vadmium
Files | <li>[crle-fix.patch](https://bugs.python.org/file37800/crle-fix.patch "Uploaded as text/plain at 2015-01-21.00:33:02 by jbeck")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2015-01-21.00:33:02.922>
labels = ['ctypes', 'performance']
title = 'ctypes.util.find_library needlessly call crle on Solaris'
updated_at = <Date 2017-01-28.22:41:17.128>
user = 'https://bugs.python.org/jbeck'
```

bugs.python.org fields:
```python
activity = <Date 2017-01-28.22:41:17.128>
actor = 'martin.panter'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['ctypes']
creation = <Date 2015-01-21.00:33:02.922>
creator = 'jbeck'
dependencies = []
files = ['37800']
hgrepos = []
issue_num = 23287
keywords = ['patch']
message_count = 6.0
messages = ['234417', '247062', '247067', '247069', '286418', '286420']
nosy_count = 4.0
nosy_names = ['swalker', 'martin.panter', 'jquast', 'jbeck']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'patch review'
status = 'open'
superseder = None
type = 'resource usage'
url = 'https://bugs.python.org/issue23287'
versions = ['Python 2.7', 'Python 3.4']
```

</p></details>

**Comment by arhadthedev** [comment]
@jcea (as a Solaris expert)

**Comment by kulikjak** [comment]
Hi, I just wanted to mention that the proposed patch is being used on Oracle Solaris in all Python versions we shipped since 2.7.
https://github.com/oracle/solaris-userland/blob/master/components/python/python311/patches/14-default-lib-path.patch

And it's not just us - OpenSolaris forks like OpenIndiana and SmartOS do the same thing for years now:
https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/python/python39/patches/14-default-lib-path.patch
https://github.com/NetBSD/pkgsrc/blob/trunk/lang/python39/patches/patch-Lib_ctypes_util.py (https://github.com/NetBSD/pkgsrc/commit/42f3d7b6779066975a94c6fe5e3b7a20db85153f)

To be fair, I cannot comment much on the possibility of users to customize library directories and potential issues with that, but it wasn't an issue in those years we all used it.

**Comment by encukou** [comment]
@kulikjak, do you want to send the patch as a PR? It simplifies the code, and I see it says “This patch was developed in-house”. I can merge it if it's submitted by an Oracle employee who signed the CLA :)


Solaris is not a supported platform per PEP-11, so I'll close the issue. A PR for it can still be merged.
Like other issues for unsupported platforms, it'll be listed on [the GitHub project](https://github.com/orgs/python/projects/27) for anyone who wants to support the platform unofficially or bring support back.
